packages/core/src/codewhisperer/util/supplementalContext/crossFileContextUtil.ts

```diff
@@ -140,6 +141,11 @@ export async function fetchSupplementalContextForSrcV1(
             content: chunk.nextContent,
             score: chunk.score,
         })
+        totalLength += chunk.nextContent.length
+
+        if (totalLength > crossFileContextConfig.maximumTotalLength) {
+            break
+        }
```

Comment:
I think this is the same result as before, we are still pushing before checking.
Move the `supplementalContexts.push` to after this check.